components/net/image_cache_thread.rs

```diff
@@ -258,6 +258,8 @@ struct ImageCache {
     // Images that are loading over network, or decoding.
     pending_loads: AllPendingLoads,
 
+    final_url: Option<ServoUrl>,
```

Comment:
This needs to be part of the [PendingLoad](https://dxr.mozilla.org/servo/rev/355d5f89da8fef0e8a2aaa94c45c2ff209e5ddae/components/net/image_cache.rs#277) and [CompleteLoad](https://dxr.mozilla.org/servo/rev/355d5f89da8fef0e8a2aaa94c45c2ff209e5ddae/components/net/image_cache.rs#196) structs instead, since each load will have a different final url value.